main.go

```diff
@@ -126,6 +129,15 @@ func main() {
 				DisableFor: []client.Object{&v1beta1.AuthorizationPolicy{}},
 			},
 		},
+		Cache: cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
+				&v1.Secret{}: {
+					Label: labels.SelectorFromSet(labels.Set{
+						"opendatahub.io/managed": "true",
```

Comment:
We should document the usage of this annotation, with this PR you are making it part of the contract: if the secrets has no annotation it will not be monitored.
I just want to make sure we keep track of the usage of this because it can easily break given that it is an impl detail.

@zdtsw @VaishnaviHire 
Do you have any advice here? Is the usage of this annotation documented/part of the "API" of the operator?

Reply:
i think we only used annotation opendatahub.io/managed in Operator, not label. and it was introduced when we added support for RawDeployment.
I dont think this annotation has been documented anywhere

Reply:
Should we be using `app.kubernetes.io/managed-by` for this instead?
https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-managed-by

Reply:
@danielezonca @zdtsw  With Selector approach we could only apply filters overs Labels and Fields. It doesn't support selector over annotations. For this approach to work we need to make sure all secrets which we want to watch should have `opendatahub.io/managed: true` label.

IF we don't follow the standard approach to add `opendatahub.io/managed: true` label then we need to think about using namespace selector.

Reply:
> Should we be using `app.kubernetes.io/managed-by` for this instead? https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-managed-by

`app.kubernetes.io/managed-by` is not maintined in our application.

Reply:
@danielezonca  I think we are a bit derived from original PR.  if the plan is to use a label on the resources (specific in this case for CM and secret) then it is not the same thing in Operator, even they are set to the same key-value (opendatahub.io/managed: true). 

@vaibhavjainwiz i am not sure by introducing `opendatahub.io/managed` here.  is it only intended for modelmesh(serving)'s resources or is planned to extend to all ODH components later(like an unified way across platform). if it is the former one, isnt `app.kubernetes.io/part-of:model-mesh` or `app.opendatahub.io/model-mesh: 'true' `labels enough? 

@jwforres we have a draft https://github.com/opendatahub-io/architecture-decision-records/pull/26 on the labeling but did not get work continued.  :cry:

Reply:
>@vaibhavjainwiz i am not sure by introducing opendatahub.io/managed here. is it only intended for modelmesh(serving)'s resources or is planned to extend to all ODH components later(like an unified way across platform). if it is the former one, isnt app.kubernetes.io/part-of:model-mesh or app.opendatahub.io/model-mesh: 'true' labels enough?

OOM in operator is common issue in almost all operator that we are facing with a client which have 2000 namespace and 49809 secrets. I had started a [doc](https://docs.google.com/document/d/1vr5Q6jn_PNYaNBW69ju5TaMHOhzCl337-GM1fBR8biU/edit#heading=h.mf8y6r2wogks) to describe this problem and its solutions. We have stated to fix this issue with **odh-model-controller** first and their share our finding with other team.

odh-model-controller query the Secret which are not labeled with `app.kubernetes.io/part-of:model-mesh`. Like Data Connection secret which is created by dashboard.

Reply:
> We should document the usage of this annotation, with this PR you are making it part of the contract:

FWIW, it seems that this "contract" has been there for a while, because the watchers are already doing filtering by this label at client side. See: https://github.com/opendatahub-io/odh-model-controller/blob/ead3013973795f8a6965d69938e0ef3ae2a1c1e0/controllers/storageconfig_controller.go#L177-L197

That seems to introduced by the end of 2022: https://github.com/opendatahub-io/odh-model-controller/commit/de166dc8a63171661abe1ee139d0c75b266ca79c. So, I think limiting the cache using that same label won't make any difference to usability. Indeed, it looks like there is a KB article in Red Hat pointing the requirement of such labels: https://access.redhat.com/solutions/7067310.

We may want to later review if we should be doing such filtering, but given current state I think this PR is on the right track.

What I think this PR is not taking into account is that odh-model-controller is creating subordinate secrets based on the watched ones. Such secrets do not have the `opendatahub.io/managed` labels, thus healing won't work. So, @vaibhavjainwiz I think you should review that... IDK if healing works currently. If it doesn't work, well, we may be good with this PR. But if it works, I'd see it as a regression.

Reply:
@israel-hdez **odh-model-controller** creates the `storage-config` secret which include `opendatahub.io/managed` label by watching dataConnectionSecretsList. 

https://github.com/opendatahub-io/odh-model-controller/blob/1ab78dfe482ce1b74b58fbe19c7e344ca69298a1/controllers/storageconfig_controller.go#L120

Reply:
> @israel-hdez **odh-model-controller** creates the `storage-config` secret which include `opendatahub.io/managed` label by watching dataConnectionSecretsList.

...good! Then I don't see anything wrong with your PR.

Reply:
> return labels["opendatahub.io/managed"] == "true" 

For platform resources, it is ok but why don't we have another label for modelserving team?  something like

```
return labels["opendatahub.io/component"] == "modelserving" 
```

one more label can give us some flexibility to work with.

Reply:
> > return labels["opendatahub.io/managed"] == "true"
> 
> For platform resources, it is ok but why don't we have another label for modelserving team? something like
> 
> ```
> return labels["opendatahub.io/component"] == "modelserving" 
> ```
> 
> one more label can give us some flexibility to work with.

@Jooho nice thought. Could you please comment it on https://github.com/opendatahub-io/architecture-decision-records/pull/26 ADR. Its for defining unified way of labeling for RHOAI.